defrag/modules/news_reddit_twitter.py

```diff
@@ -0,0 +1,193 @@
+# Defrag - centralized API for the openSUSE Infrastructure
+# Copyright (C) 2021 openSUSE contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+from attr import dataclass
+from defrag.modules.helpers.sync_utils import as_async
+from os import environ as env
+from typing import Any, Deque, List, NamedTuple, Optional
+import aiohttp
+from aiohttp.client import ClientSession
+from datetime import date, datetime
+from collections import namedtuple, deque
+from operator import attrgetter
+import atoma
+import asyncio
+import twitter
+
+""" INFO
+This module provides a class -- subclasses 2x below -- to
+serve as querying & storing point for recent reddit posts (r/openSUSE)
+and tweets (sent by user "@openSUSE"). Additional features could be:
+- posting (would require higher credentials)
+- fetching also Tweets *about* openSUSE and also Reddit comments 
+to model data answering the question: "What are people talking about recently?"
+"""
+
+""" TO DO
+- Remove 'Req' because it will be used elsewhere  
+- Add Reddis caching
+"""
+__MOD_NAME__ = "news_reddit_twitter"
+
+
+class Req:
+    """ Simple context manager to manage async HTTP requests. """
+
+    session: Optional[ClientSession] = None
+
+    @classmethod
+    def open_session(cls):
+        if not cls.session:
+            cls.session = aiohttp.ClientSession()
+
+    @classmethod
+    async def close_session(cls):
+        if cls.session and not cls.session.closed:
+            await cls.session.close()
+
+    def __init__(self, url: str, closeOnExit: bool = True):
+        if not self.session:
+            self.open_session()
+        self.url = url
+        self.closeOnExit = closeOnExit
+
+    async def __aenter__(self):
+        return await self.session.get(self.url)
```

Comment:
This way it fails when behind a proxy, because aiohttp does not use the system proxy for whatever reason. I'd suggest something like this:
```suggestion
        return await self.session.get(self.url, proxy=env["http_proxy"])
```

Reply:
Does my response on line 54 of the file address your recommendation?